src/credentials/CHIPOpCred.h

```diff
@@ -0,0 +1,184 @@
+     *        allocated externally and methods in this class don't need to deal with memory allocations.
+     *
+     * @param certSetsArray         A pointer to the array of the ChipCertificateSet structures.
+     * @param certSetsArraySize     Number of ChipCertificateSet entries in the array.
+     * @param deviceOpCredKeypair   A pointer to the Device Operational Credential's corresponding Keypair.
+     * @param deviceCredentialsDER  A pointer to the raw data buffer in DER format for the Device Operational Credentials.
+     * @param deviceCredentialsLen  Length in bytes of the Device Operational Credentials.
+     *
+     * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR Init(ChipCertificateSet * certSetsArray, uint8_t certSetsArraySize, P256Keypair * deviceOpCredKeypair,
+                    const uint8_t * deviceCredentialsDER, uint16_t deviceCredentialsLen);
+
+    /**
+     * @brief Find certificate set in the set.
+     *
+     * @param trustedRootId  Authority key identifier of the root certificate-set to be found in the operational credential set.
+     *
+     * @return A pointer to the certificate data On success. Otherwise, NULL if no certificate found.
+     **/
+    ChipCertificateSet * FindCertSet(const CertificateKeyId & trustedRootId) const;
+
+    /**
+     * @return A pointer to the set of certificate-set data entries.
+     **/
+    const ChipCertificateSet * GetOpCredSet() const { return mOpCreds; }
+
+    /**
+     * @return A pointer to the last certificate-set in the set. Returns NULL if certificate set is empty.
+     **/
+    const ChipCertificateSet * GetLastCert() const { return (mOpCredCount > 0) ? &mOpCreds[mOpCredCount - 1] : nullptr; }
+
+    /**
+     * @return Number of certificate-sets loaded into the set.
+     **/
+    uint8_t GetCertCount() const { return mOpCredCount; }
+
+    /**
+     * @brief Retrieve Trusted Root ID / Authority Key ID of a Certificate Set.
+     *
+     * @param certSetIndex  Index of the Certificate-set to be used to search for a Trusted Root ID.
+     *
+     * @return A pointer to the Trusted Root ID on success. Otherwise, nullptr if no Trust Anchor is found.
+     **/
+    const CertificateKeyId * GetTrustedRootId(uint16_t certSetIndex) const;
+
+    /**
+     * @brief Check whether certificate set is in the operational credential set.
+     *
+     * @param cert  Pointer to the ChipCertificateSet structures.
+     *
+     * @return True if certificate is in the set, false otherwise.
+     **/
+    bool IsCertSetInTheOpCredSet(const ChipCertificateSet * cert) const;
+
+    /**
+     * @brief Check whether Trusted Root ID is in the operational credential set.
+     *
+     * @param trustedRoot  Reference to the Trusted Root ID.
+     *
+     * @return True if Trusted Root ID is in the set, false otherwise.
+     **/
+    bool IsTrustedRootIn(const CertificateKeyId & trustedRoot) const;
+
+    /**
+     * @brief Validate CHIP certificate.
```

Comment:
This @Brief restates the code: `ValidateCert` obviously validates the certificate (and we are in project chip, so 'CHIP certificate' also seems implied.

Could we:
  - replace abbreviations throughout: `ValidateCertificate` to favor readability over typing speed
  - add doccomments that state things not obvious. In this particular example, I wonder 'What is a valid cip certificate?'